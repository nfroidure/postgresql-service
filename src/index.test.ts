import { describe, beforeEach, test, jest, expect } from '@jest/globals';
import assert from 'assert';
import { Knifecycle, constant } from 'knifecycle';
import initPGService from './index.js';
import type { PGService } from './index.js';
import type { LogService } from 'common-services';

describe('initPGService', () => {
  const log = jest.fn<LogService>();
  const PG = {};

  beforeEach(() => {
    log.mockClear();
  });

  describe('pg', () => {
    test('should work without envname', async () => {
      const pg = await initPGService({
        PG: { ...PG, password: 'a_pwd' },
        log,
      });

      assert(pg);
      expect(log.mock.calls).toMatchSnapshot();
    });

    test('should work with envname', async () => {
      const pg = await initPGService({
        PG_URL_ENV_NAME: 'YOLO',
        ENV: {
          YOLO: 'myurl',
        },
        PG,
        log,
      });

      assert(pg);
      log.mockClear();

      expect(log.mock.calls).toMatchSnapshot();
    });
  });

  test('should work with Knifecycle', async () => {
    const $ = new Knifecycle()
      .register(initPGService)
      .register(constant('log', log))
      .register(constant('PG', {}))
      .register(constant('ENV', { PG_URL: 'a_pg_url' }));
    const { pg } = await $.run<{ pg: PGService }>(['pg']);

    assert(pg);
    expect(log.mock.calls).toMatchSnapshot();
  });
});
