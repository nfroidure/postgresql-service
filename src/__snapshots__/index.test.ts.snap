// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initPGService pg should work with envname 1`] = `[]`;

exports[`initPGService pg should work without envname 1`] = `
[
  [
    "warning",
    "‚ö†Ô∏è - Setting the password in the PG config is unsafe.",
  ],
  [
    "warning",
    "üêò - Initializing PG service...",
  ],
]
`;

exports[`initPGService should work with Knifecycle 1`] = `
[
  [
    "warning",
    "üêò - Initializing PG service...",
  ],
]
`;
